src/systems/diffeqs/abstractodesystem.jl

```
@@ -555,12 +640,24 @@ function DiffEqBase.ODEProblem{iip}(sys::AbstractODESystem,u0map,tspan,
                                     parammap=DiffEqBase.NullParameters();kwargs...) where iip
     has_difference = any(isdifferenceeq, equations(sys))
     f, u0, p = process_DEProblem(ODEFunction{iip}, sys, u0map, parammap; has_difference=has_difference, kwargs...)
-    if has_difference
-        ODEProblem{iip}(f,u0,tspan,p;difference_cb=generate_difference_cb(sys;kwargs...),kwargs...)
+    if has_continuous_events(sys)
+        event_cb = generate_rootfinding_callback(sys; kwargs...)
+    else
+        event_cb = nothing
+    end
+    difference_cb = has_difference ? generate_difference_cb(sys; kwargs...) : nothing
+    cb = merge_cb(event_cb, difference_cb)
+
+    if cb === nothing
+        ODEProblem{iip}(f, u0, tspan, p; kwargs...)
     else
-        ODEProblem{iip}(f,u0,tspan,p;kwargs...)
+        ODEProblem{iip}(f, u0, tspan, p; callback=cb, kwargs...)

```

Let's check that `haskey(kwargs, :callback)` is false. Otherwise either user's input is silently ignored, or our generated callback is silently ignored.
```julia
julia> foo(;a) = a
foo (generic function with 1 method)

julia> kwargs = (a = 100,)
(a = 100,)

julia> foo(;a=10, kwargs...)
100
```

---

I merged user-provided callbacks with automatically generated ones